cache/engine/client/client.go

```diff
@@ -105,12 +107,30 @@ func (rdb *RedisClient) unlock(ctx context.Context, key string) error {
 
 // PurgeAll - Purges all the existing keys on a DB.
 func (rdb *RedisClient) PurgeAll() (bool, error) {
+	if rdb.Client.ClusterSlots(ctx).Err() == nil {
+		clusterClient := rdb.Client.(*goredislib.ClusterClient)
+
+		err := clusterClient.ForEachShard(ctx, func(ctx context.Context, client *goredislib.Client) error {
+			return rdb.purgeAllClusterKeys(client)
+		})
+
+		return err == nil, err
+	} else {
+		_, err := circuitbreaker.CB(rdb.Name, rdb.logger).Execute(func() (interface{}, error) {
+			err := rdb.Client.FlushDB(ctx).Err()
+			return nil, err
+		})
+
+		return err == nil, err
+	}
+}
+
+func (rdb *RedisClient) purgeAllClusterKeys(client *goredislib.Client) error {
```

Comment:
reducing similar code:

```suggestion
	// multiple redis instances
	if rdb.Client.ClusterSlots(ctx).Err() == nil {
		clusterClient := rdb.Client.(*goredislib.ClusterClient)

		err := clusterClient.ForEachShard(ctx, func(ctx context.Context, client *goredislib.Client) error {
			return rdb.purgeAllKeys(client)
		})

		return err == nil, err
	}
	
	// single redis instance
	err := rdb.purgeAllKeys(rdb.Client)
	return err == nil, err
}

func (rdb *RedisClient) purgeAllKeys(client *goredislib.Client) error {
```

Reply:
Good point, done!